src/tests/conformance/messages.rs

```diff
@@ -564,67 +564,121 @@ async fn eagerly_crawls_network_for_peers() {
+    //          b) Query GetAddr
+    //          c) Receive Addr == N peer addresses
     //
-    // This test currently fails for zcashd and zebra passes.
+    // This test currently fails for both zcashd and zebra.
     //
     // Current behaviour:
     //
     //  zcashd: Never responds. Logs indicate `Unknown command "getaddr" from peer=1` if we initiate
     //          the connection. If the node initiates the connection then the command is recoginized,
     //          but likely ignored (because only the initiating node is supposed to send it).
     //
-    //  zebra:  Infinitely spams `GetAddr` and `GetData`. Can be coaxed into responding correctly if
-    //          all its peer connections have responded to `GetAddr` with a non-empty list.
+    //  zebra:  Times out due to infinitely spamming `GetAddr` and `GetData`.
+    //
+    //          Can be coaxed into responding by sending a non-empty Addr in
+    //          response to node's GetAddr. This still fails as it includes previous inbound
+    //          connections in its address book (as in the bug listed above).
+    //
 
-    // Create a node and main connection
+    // Establish N listener peer nodes
+    const PEERS: usize = 5;
+    let mut peers = Vec::with_capacity(PEERS);
+    for _ in 0..PEERS {
+        peers.push(TcpListener::bind(new_local_addr()).await.unwrap())
+    }
+    let mut peer_addrs = peers
+        .iter()
+        .map(|peer| peer.local_addr().unwrap())
+        .collect::<Vec<_>>();
+    peer_addrs.sort_unstable();
+    let mut peer_handles = Vec::with_capacity(peers.len());
+    for peer in peers {
+        peer_handles.push(tokio::spawn(async move {
+            peer.accept().await.unwrap();
+        }));
+    }
+
+    // Start node with an initial set of peers to connect to
     let mut node: Node = Default::default();
-    node.initial_action(Action::WaitForConnection(new_local_addr()))
+    node.initial_action(Action::None)
+        .initial_peers(peer_addrs.clone())
         .start()
         .await;
 
-    let mut peers = Vec::new();
+    // wait for all peers to get connected to
+    for handle in peer_handles {
+        handle.await.unwrap();
+    }
 
-    for i in 0u8..5 {
-        let mut new_peer = initiate_handshake(node.addr()).await.unwrap();
-        let filter = MessageFilter::with_all_auto_reply().enable_logging();
+    // List of inbound peer addresses (from the nodes perspective).
+    // This is used to test that the node does not gossip these addresses.
+    // (this exercises a known Zebra bug: https://github.com/ZcashFoundation/zebra/pull/2120)
+    let mut inbound = Vec::new();
 
-        Message::GetAddr
-            .write_to_stream(&mut new_peer)
-            .await
-            .unwrap();
+    // Connect to node and request GetAddr.
+    // We perform multiple iterations in order to exercise the above Zebra bug.
+    for i in 0..5 {
+        let mut stream = initiate_handshake(node.addr()).await.unwrap();
```

Comment:
the version in `initiate_handshake` might need to send the listener address?

Reply:
I'm honestly not sure. In our current test setup I think it doesn't matter - because we are on a local network which bitcoin terms as "unroutable". From the second bitcoin example for [version](https://en.bitcoin.it/wiki/Protocol_documentation#version)

` notice that it does not fill out the address information at all when the source or destination is "unroutable".`

And the send/recv address are simply filled out identically:
```
 01 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 FF FF 00 00 00 00 00 00 - Recipient address info - see Network Address
 01 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 FF FF 00 00 00 00 00 00 - Sender address info - see Network Address
```

Reply:
Fair enough, in normal circumstances, the node might expect to find the peer's listener there. This will get refactored when adding `pea2pea` so no need to fix this now.